uacp/conn.go

```diff
@@ -36,11 +38,18 @@ func nextid() uint32 {
 
 func Dial(ctx context.Context, endpoint string) (*Conn, error) {
 	debug.Printf("Connect to %s", endpoint)
-	network, raddr, err := ResolveEndpoint(endpoint)
+
+	ctx, cancel := context.WithTimeout(ctx, DefaultDialTimeout)
```

Comment:
Currently the only way would be to set `uacp.DefaultDialTimeout` first. What I'd rather have is to be able to use a context with timeout here: https://github.com/Intelecy/opcua/blob/net-ctx/client.go#L158-L189.
But that requires more refactoring as the incoming context is used for the monitoring routine.

Reply:
Right, but `uacp.DefaultDialTimeout` is `const` so it can't be configured at runtime.  Changing it to be non-const means users would need to import `uacp` and modify a global variable to change the setting.

I agree that putting it in the client is the right spot which is why I think having `uacp.Dial` just accept a timeout parameter (or making a `uacp.DialWithTimeout` method and have `Dial` be a wrapper for it with the default timeout in place) could work.  Then the client's `Dial` method just changes to:
`conn, err := uacp.DialWithTimeout(ctx, c.endpointURL, c.cfg.DialTimeout)` and your changes still work (replacing `DefaultDialTimeout` with the parameter)

Alternatively, it could be possible to create a second context in the the client's `Dial` method with a timeout as long as it's not used as the parent for the monitor routine's context and that shouldn't cause issues. 
eg:
```go
func (c *Client) Dial(ctx context.Context) error {
...

    dialCtx, dialCancel := context.WithTimeout(ctx, c.cfg.DialTimeout)
    defer dialCancel()
    conn, err := uacp.Dial(dialCtx, c.endpointURL)
    if err != nil {
	    return err
    }

...

    ctx, c.cancelMonitor = context.WithCancel(ctx)
    go c.monitorChannel(ctx)

...
```
While this doesn't seem as clear to me, it might make having no timeout simpler.